### ISSUE TITLE
Multiple DefaultHttpClientFactory container registrations

### ISSUE THREAD

**Issue by gravity00** [aspnet/Extensions#1679]
It seems the class `DefaultHttpClientFactory` is being registered multiple times into the container because the base `AddHttpClient` uses the instruction `AddSingleton<DefaultHttpClientFactory>();` instead of `TryAddSingleton<DefaultHttpClientFactory>();`.

https://github.com/aspnet/Extensions/blob/52b4a202090891dd254ae83ab97b01a99b5f0eb9/src/HttpClientFactory/Http/src/DependencyInjection/HttpClientFactoryServiceCollectionExtensions.cs#L35-L38

I believe this isn't an intended behavior and will cause multiple registrations when using the example code provided by the [documentation](https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests#how-to-use-typed-clients-with-httpclientfactory):

```csharp
// Startup.cs
//Add http client services at ConfigureServices(IServiceCollection services) 
services.AddHttpClient<ICatalogService, CatalogService>();
services.AddHttpClient<IBasketService, BasketService>();
services.AddHttpClient<IOrderingService, OrderingService>();
```

**Comment by rynowak** [comment]
Sure. In practice this isn't a problem unless you retrieve `DefaultHttpClientFactory` an an `IEnumerable<>` but I see no reason why not to do the most correct thing :+1:

**Comment by rynowak** [comment]
Fixed by #1682
